spec/api.yaml

```diff
@@ -0,0 +1,269 @@
+        '404':
+          description: The table doesn't exist
+        '500':
+          description: An unexpected error occurred
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+  /search:
+    post:
+      summary: Perform a search on Tables
+      description: Optional operation that accepts a SearchRequest and returns a TableData
+      operationId: search
+      requestBody:
+        description: Query to execute
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/SearchRequest"
+      tags:
+        - Search API
+      responses:
+        '200':
+          description: Query results returned as TableData
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/TableData"
+        '400':
+          description: The query is invalid
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        '404':
+          description: This server does not implement the search operation
+        '500':
+          description: An unexpected error occurred
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+  /service-info:
+    get:
+      summary: 'Show information about the Data Connect service'
+      operationId: getServiceInfo
+      tags:
+        - Service Info API
+      responses:
+        '200':
+          description: |
+            Show information about this Data Connect service.
+
+            Use `"type": {"group": "org.ga4gh", "artifact": "data-connect", "version": "0.1.0"}` when implementing this specification directly.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+components:
+  securitySchemes:
+    bearerAuth:
+      type: http
+      scheme: bearer
+  schemas:
+    ListTablesResponse:
+      required:
+        - tables
+      type: object
+      properties:
+        tables:
+          type: array
+          items:
+            $ref: "#/components/schemas/Table"
+        pagination:
+          $ref: '#/components/schemas/Pagination'
+        errors:
+          $ref: "#/components/schemas/ErrorList"
+    Table:
+      required:
+        - name
+        - data_model
+      type: object
+      properties:
+        name:
+          type: string
+          description: Table name
+        description:
+          type: string
+          description: Optional description of the Table
+        data_model:
+          $ref: "http://json-schema.org/draft-07/schema#"
+        errors:
+          $ref: "#/components/schemas/ErrorList"
+      description: |
+        Describes a Table hosted by this Data Connect node.
+    TableData:
+      required:
+        - data_model
```

Comment:
We should remove this. The `data_model` is only required here if the `data` array is present. This was a late change to allow servers to implement "status polling via pagination". Should also add description text about this.